hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFileSystem.java

```diff
@@ -1940,7 +1940,8 @@ public Path getEnclosingRoot(Path path) throws IOException {
       try {
         res = fsState.resolve((path.toString()), true);
       } catch (FileNotFoundException ex) {
-        throw new NotInMountpointException(path, String.format("getEnclosingRoot - %s", ex.getMessage()));
+        throw new NotInMountpointException(path,
+            String.format("getEnclosingRoot - %s", ex.getMessage()));
```

Comment:
ok, add an initCause(ex) building this exception and the one in viewFS. stack traces are too useful to throw away

Reply:
Added, thanks!